packages/web3-core-requestmanager/src/index.ts

```diff
@@ -0,0 +1,51 @@
+import Web3ProviderHttp from 'web3-providers-http';
+import { ProviderOptions } from 'web3-providers-base/types';
+
+// TODO Make eslint happy
+/* eslint-disable */
+enum ProviderProtocol { UNKNOWN, HTTP, WS, IPC }
+
+export default class Web3RequestManager {
+    provider: Web3ProviderHttp | undefined
+
+    providerProtocol: ProviderProtocol = ProviderProtocol.UNKNOWN
+
+    constructor(providerOptions: ProviderOptions) {
+      switch (Web3RequestManager.detectProviderProtocol(providerOptions.providerString)) {
+        case ProviderProtocol.HTTP:
+          this.provider = new Web3ProviderHttp(providerOptions);
+          this.providerProtocol = ProviderProtocol.HTTP;
+          break;
+        case ProviderProtocol.WS:
+          this.providerProtocol = ProviderProtocol.WS;
+          // TODO
+          break;
+        default:
+          // TODO figure out if possible to support generic provider
+          break;
+      }
+    }
+
+    static detectProviderProtocol(providerString: string): ProviderProtocol {
+      try {
+        if (/^http(s)?:\/\//i.test(providerString)) {
+          return ProviderProtocol.HTTP;
+        } if (/^ws(s)?:\/\//i.test(providerString)) {
+          return ProviderProtocol.WS;
+        }
+        return ProviderProtocol.UNKNOWN;
+      } catch (error) {
+        throw Error(`Error detecting provider protocol: ${error.message}`);
+      }
+    }
+
+    // TODO get rid of anys
+    async send(options: any): Promise<any> {
```

Comment:
I'm assuming since we're not supporting the callback scheme, and forcing promises we can get away with removing all the old boilerplate from the request manager found [here](https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-core-requestmanager/src/index.js#L167-L190)

Reply:
Yes, but I'm not so sure we can actually get away with this. I really hope so, because it's a heck of a lot of code that just clutters everything, but I imagine projects that use older JS aren't going to be happy

Reply:
1.x will be LTS for long enough that they can upgrade. :)